### DISCUSSION TITLE
[POQ] How to specify values for product specification

### DISCUSSION THREAD

**Discussion by mariusz-amartus** [MEF-GIT/MEF-LSO-Sonata-SDK#173] [Team Posts]
Hi,

Here's the question I received (LSO Sonata SDK R3):

> 2. How can we specify the values for product specification (e.g. Bandwidth=10M) in POQ?
> In the swagger, it only accept the product specification ID and the JSON schema:
> ![swagger_product_spec](https://user-images.githubusercontent.com/52410146/66313620-f45e1380-e912-11e9-8882-bc283a0c9676.png)

According to my understanding:

> Bandwidth profile could be specified in the product specification which needs to be included for code generation. For E-Line please see:
> https://github.com/MEF-GIT/MEF-LSO-Sonata-SDK/blob/working-draft/payload_descriptions/ProductSpecDescription/MEF_ELineSpec_v3.json
> 
> ```
> 		"ENNIIngressBWProfile":{
>          "description":"The ingress bandwidth profile at ENNI",
>          "items":{
>             "$ref":"#/definitions/BandwidthProfile"
>          }
> 		},
> 		"UNIIngressBWProfile":{
>          "description":"The ingress bandwidth profile at UNI",
>          "items":{
>             "$ref":"#/definitions/BandwidthProfile"
>          }
> 		},
> 		"BandwidthProfile":{
>          "description":"",
>          "type":"object",
>          "properties":{
>             "cir":{
>                "description":"As defined in MEF 26.2 for ENNI & 10.3 for UNI, this value is expressed via Information rate.",
>                "$ref":"#/definitions/informationRate"
>             },
>             "cbs":{
>                "description":"As defined in MEF 26.2 for ENNI & 10.3 for UNI, this value is expressed as bytes",
>                "type":"integer",
> 			   "format":"int32"
>             },
>             "eir":{
>                "description":"As defined in MEF 26.2 for ENNI & 10.3 for UNI, this value is expressed via Information rate.",
>                "$ref":"#/definitions/informationRate"
>             },
>             "ebs":{
>                "description":"As defined in MEF 26.2 for ENNI & 10.3 for UNI, this value is expressed as bytes",
>                "type":"integer",
> 			   "format":"int32"
>             },
>             "colorMode":{
>                "description":"This enumeration indicates whether the Color Identifier of the Service Frame is considered by the Bandwidth Profile Algorithm",
>                "type":"string",
> 			   "enum":[
> 					"COLOR_AWARE",
> 					"COLOR_BLIND"
> 				]
>             },
>             "couplingFlag":{
>                "description":"The Bandwidth Profile parameter that determines whether overflow tokens not used for Service Frames declared Green can be used as Yellow tokens.",
>                "type":"boolean"
>             }
>          }
>       }
> ```

@bartoszm, @bigludo7, others,
Could you please comment on the above?

Thanks,
Mariusz
LSO SDK Community Manager

**Comment by bartoszm** [comment]
I have prepared a short blog entry describing the current situation (or rather my opinion on the current situation).
By no means this is the best or the only way to address this.
I hope, though, it might be helpful: https://amartus.com/fast-track-from-sonata-api-definition-to-running-code/
